Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+        trackLayer.timeSlider = self
+        trackLayer.contentsScale = UIScreen.main.scale
+        layer.addSublayer(trackLayer)
+        
+        // Add tick layer
+        tickMarkLayer.timeSlider = self
+        tickMarkLayer.contentsScale = UIScreen.main.scale
+        layer.addSublayer(tickMarkLayer)
+        
+        // Add lower thumb layer
+        lowerThumbLayer.timeSlider = self
+        lowerThumbLayer.contentsScale = UIScreen.main.scale
+        layer.addSublayer(lowerThumbLayer)
+        
+        // Add upper thumb layer
+        upperThumbLayer.timeSlider = self
+        upperThumbLayer.contentsScale = UIScreen.main.scale
+        layer.addSublayer(upperThumbLayer)
+        
+        // Add the minimum value label
+        fullExtentStartTimeLabel.isHidden = !(fullExtentLabelsVisible && isSliderVisible)
+        fullExtentStartTimeLabel.foregroundColor = fullExtentLabelColor.cgColor
+        fullExtentStartTimeLabel.alignmentMode = kCAAlignmentCenter
+        fullExtentStartTimeLabel.frame = CGRect.zero
+        fullExtentStartTimeLabel.contentsScale = UIScreen.main.scale
+        fullExtentStartTimeLabel.fontSize = fullExtentLabelFont.pointSize
+        layer.addSublayer(fullExtentStartTimeLabel)
+        
+        // Add the minimum value label
+        fullExtentEndTimeLabel.isHidden = !(fullExtentLabelsVisible && isSliderVisible)
+        fullExtentEndTimeLabel.foregroundColor = fullExtentLabelColor.cgColor
+        fullExtentEndTimeLabel.alignmentMode = kCAAlignmentCenter
+        fullExtentEndTimeLabel.frame = CGRect.zero
+        fullExtentEndTimeLabel.contentsScale = UIScreen.main.scale
+        fullExtentEndTimeLabel.fontSize = fullExtentLabelFont.pointSize
+        layer.addSublayer(fullExtentEndTimeLabel)
+        
+        // Add the lower value label
+        currentExtentStartTimeLabel.isHidden = !(labelMode == .thumbs && isSliderVisible)
+        currentExtentStartTimeLabel.foregroundColor = currentExtentLabelColor.cgColor
+        currentExtentStartTimeLabel.alignmentMode = kCAAlignmentCenter
+        currentExtentStartTimeLabel.frame = CGRect.zero
+        currentExtentStartTimeLabel.contentsScale = UIScreen.main.scale
+        currentExtentStartTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        layer.addSublayer(currentExtentStartTimeLabel)
+        
+        // Add the upper value label
+        currentExtentEndTimeLabel.isHidden = !(labelMode == .thumbs && isSliderVisible)
+        currentExtentEndTimeLabel.foregroundColor = currentExtentLabelColor.cgColor
+        currentExtentEndTimeLabel.alignmentMode = kCAAlignmentCenter
+        currentExtentEndTimeLabel.frame = CGRect.zero
+        currentExtentEndTimeLabel.contentsScale = UIScreen.main.scale
+        currentExtentEndTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        layer.addSublayer(currentExtentEndTimeLabel)
+        
+        // Create the images
+        let bundle = Bundle(for: type(of: self))
```

Comment:
Constructing the following 4 images from bundle is unnecessary, no? Wouldn't `UIImage(named: "ImageName")!` for each of these 4 images suffice?

Reply:
`UIImage(named:` is not working. Hence, we are using bundle. Other toolkit components using too.

Reply:
@esreli Specifying the bundle is necessary because UIImage.init(named:) looks in `Bundle.main`, which is the app's bundle.